lib/round.rb

```diff
@@ -0,0 +1,119 @@
+
+class Round
+  attr_reader :deck, :turns, :number_correct
+  #Initializes the round object and passes an arguement deck.
+  def initialize(deck)
+    @deck = deck
+    @turns = []
+    @number_correct = 0
+
+
+  end
+
+
+  def current_card
+
+   deck.cards.first
+
+  end
+
+  def take_turn(guess)
+
+      turn = Turn.new(guess, current_card)
+        @turns << turn
+        if guess == current_card.answer
+            @number_correct += 1
+        end
+        deck.cards.rotate!
+            #may require tinkering in iteration_3
+        return turn
+
+  end
+
+
+
+  def number_correct_by_category(category)
+
+    correct_turns = turns.select do |turn|
+
+
+      turn.card.category == category && turn.correct?
+
+    end
+
+    correct_turns.count
+
+  end
+
+  def percent_correct
+
+   (number_correct.to_f/(turns.count.to_f)) * 100
+
+  end
+
+
+
+  def percent_correct_by_category(category)
+
+    percent_correct_cat = 0.0
+
+      turns.each do |turn|
+        if turn.card.category == category
+          percent_correct_cat += 1.0
+        end
+      end
+
+      (number_correct_by_category(category)/(percent_correct_cat)) * 100
+    end
```

Comment:
since this is being run after all of your cards have been iterated through, you could utilize you cards_in_category method, instead of iterating through the turns collection.